### PR TITLE
Feature plat 15

### PR DESCRIPTION
**Pull request by gimenomarc**

### CODE REVIEW

File: src/Components/Session_id/Session_id.jsx [hunk 1]
```diff
@@ -0,0 +1,10 @@
+import { v4 as uuid } from 'uuid';
+import Cookies from 'universal-cookie';
+
+const session_id = uuid();
+console.log(session_id);
+
+const daysToExpire = new Date(2147483647 * 1000);
```
Comment (EsteveSegura) [inline src/Components/Session_id/Session_id.jsx:7]: This look constant, should be defined in an static file of constants.

File: src/Components/Session_id/Session_id.jsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import { v4 as uuid } from 'uuid';
+import Cookies from 'universal-cookie';
+
+const session_id = uuid();
+console.log(session_id);
```
Comment (EsteveSegura) [inline src/Components/Session_id/Session_id.jsx:5]: console log? remove that!

File: src/Components/Session_id/Session_id.jsx [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { v4 as uuid } from 'uuid';
+import Cookies from 'universal-cookie';
+
+const session_id = uuid();
```
Comment (EsteveSegura) [inline src/Components/Session_id/Session_id.jsx:4]: this should be inside in a abstraction

**Comment by EsteveSegura** [comment]
Does not meet the criteria specified in the task.... Every time I do F5 it assigns me a new sessionid.... how can I track a user? if every time you change the session id?

File: package.json [hunk 4]
```diff
@@ -12,6 +12,8 @@
     "@testing-library/react": "^13.3.0",
     "@testing-library/user-event": "^13.5.0",
     "react": "^18.1.0",
+    "react-cookie": "^4.1.1",
+    "react-device-detect": "^2.2.2",
```
Comment (EsteveSegura) [inline package.json:16]: You're using it? uninstall that

File: src/Components/Session_id/Session_id.jsx [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+const daysToExpire = new Date(MAX_DATE);
+
+const cookies = new Cookies();
+
+const sessionIdExists = cookies.get('session_id')
+if(!sessionIdExists) {
+    cookies.set('session_id', session_id, { path: '/', expires: daysToExpire });
+}
+
+console.log(cookies)
```
Comment (EsteveSegura) [inline src/Components/Session_id/Session_id.jsx:16]: Remember, allways remove console.logs

File: src/Services/idGenerator [hunk 6]
```diff
@@ -0,0 +1,4 @@
+import { v4 as uuid } from 'uuid';
+
+export var generateId = uuid();
```
Comment (EsteveSegura) [inline src/Services/idGenerator:3]: Why using "var" ? this is forbidden

Comment (gimenomarc) [reply src/Services/idGenerator:3]: Changed to a const

File: src/Components/Session_id/Session_id.jsx (hunk 5 above)
Comment (gimenomarc) [reply src/Components/Session_id/Session_id.jsx:16]: Removed

File: package.json (hunk 4 above)
Comment (gimenomarc) [reply package.json:16]: Uninstalled

**Comment by EsteveSegura** [comment]
Why is session id logic a component of react?
According to the documentation for react.

A component is a UI element
This is not a UI component

Components are created to be reusable
This is the least reusable piece of code ever.

You should move all this logic to a more appropriate place in session_id.jsx and it should not be executed in App.js, try it in main.js.

File: src/index.js [hunk 7]
```diff
@@ -3,6 +3,7 @@ import ReactDOM from 'react-dom/client';
 import './index.css';
 import App from './App';
 import reportWebVitals from './reportWebVitals';
+import './Services/Session_id/Session_id'
```
Comment (EsteveSegura) [inline src/index.js:6]: CamelCase or SnakeCase?

Comment (gimenomarc) [reply src/index.js:6]: I'll use CamelCase in all cases. (Changed Session_id for sessionId)

**Pull request merged**